frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -64,6 +81,92 @@ export const DeckGlJsonChart: FC<PropsWithHeight> = props => {
     setIsInitialized(true)
   }, [])
 
+  const handleClick = useCallback(
+    (
+      info: PickingInfo,
+      event: { srcEvent: MouseEvent | TouchEvent | PointerEvent }
+    ) => {
+      const {
+        color,
+        index,
+        picked,
+        x,
+        y,
+        pixel,
+        coordinate,
+        devicePixel,
+        pixelRatio,
+        object,
+      } = info
+
+      const layerId = `${info.layer?.id || null}`
+      const wasShiftClick = event.srcEvent.shiftKey
+
+      const stringValue = widgetMgr.getStringValue(element)
+      const currState: DeckGlElementState | null = stringValue
+        ? JSON5.parse(stringValue)
+        : null
+
+      const indices: number[] = wasShiftClick
+        ? currState?.selection[layerId]?.indices || []
+        : []
+
+      const objects: unknown[] = wasShiftClick
+        ? currState?.selection[layerId]?.objects || []
+        : []
+
+      const selection: LayerSelection = {
+        last_selection: {
+          color,
+          layer: layerId,
+          index,
+          picked,
+          x,
+          y,
+          pixel,
+          coordinate,
+          devicePixel,
+          pixelRatio,
+          object,
+        },
+        indices,
+        objects,
+      }
+
+      const existingIndex = indices.indexOf(index)
+
+      if (wasShiftClick && existingIndex !== -1) {
+        // Unselect an existing index
+        indices.splice(existingIndex, 1)
+        objects.splice(existingIndex, 1)
```

Comment:
I was trying to find a better way to replace the parallel lists? My first thought was a set or a map (index -> object) and then using `map.keys()` and `map.values()` for cleanliness, but that might be overkill.

Reply:
Nice suggestion! I updated this to utilize a `Map`